primitives/merkle-mountain-range/src/lib.rs

```diff
@@ -351,6 +351,38 @@ impl_leaf_data_for_tuple!(A:0, B:1, C:2);
 impl_leaf_data_for_tuple!(A:0, B:1, C:2, D:3);
 impl_leaf_data_for_tuple!(A:0, B:1, C:2, D:3, E:4);
 
+/// A MMR proof data for a group of leaves.
+#[derive(codec::Encode, codec::Decode, RuntimeDebug, Clone, PartialEq, Eq)]
+pub struct BatchProof<Hash> {
+	/// The indices of the leaves the proof is for.
+	pub leaf_indices: Vec<LeafIndex>,
+	/// Number of leaves in MMR, when the proof was generated.
+	pub leaf_count: NodeIndex,
+	/// Proof elements (hashes of siblings of inner nodes on the path to the leaf).
+	pub items: Vec<Hash>,
+}
+
+impl<Hash> BatchProof<Hash> {
+	/// Will panic if leaf indices length is less than 1
+	pub fn into_single_leaf_proof(proof: BatchProof<Hash>) -> Proof<Hash> {
```

Comment:
I just noticed this and think we should do checked access and return `Error::InvalidLeafIndices` instead of unchecked access and panicking.